Improve CloudFront Cache Invalidation Retry

sureshc opened the pull request:
The automated build process on our managed systems (staging, test, production, etc.) are more frequently encountering internal service errors when invoking the CloudFront cache invalidation API. The on call engineer typically manually restarts the build to workaround the issue. By default, the AWS Ruby v3 SDK retries this API call 3 times, with likely no significant delay between retries. Switch from this `legacy` default retry configuration to the newer `standard` retry algorithm and increase the number of retries to reduce the likelihood of these service errors disrupting the build on a long running managed system.

Also - remove / rename outdated components now that we no longer use Varnish and don't need to flush its cache.

## Links

https://codedotorg.atlassian.net/browse/INF-717

## Testing story

There are no unit tests on this logic. Should we add some here?
In the absence of unit tests, we can provision an adhoc with CloudFront enabled and execute the cache invalidation rake task:

`bundle exec rake adhoc:cdn:start RAILS_ENV=adhoc STACK_NAME=adhoc-flush-cache`

```
ubuntu@adhoc-flush-cache:~/adhoc$ bundle exec rake ci:flush_cache
Creating CloudFront cache invalidations...
Invalidations created: 2

ubuntu@adhoc-flush-cache:~/adhoc$ bin/flush_cache
Flushing CloudFront caches...
Creating CloudFront cache invalidations...
Invalidations created: 2
All done!
```

<img width="1261" alt="image" src="https://user-images.githubusercontent.com/2157034/191629499-078115b2-d084-4aef-b26d-7afea213dedf.png">



## Deployment strategy

## Follow-up work

Investigate (possibly with AWS Support) why we are encountering this error with increasing frequency. Is it because of the large number of Cache Behaviors we have configured on each CloudFront Distribution? We have been granted a higher number of Behaviors than is normally permitted on a Distribution.

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

cat5inthecradle commented on bin/flush_cache:
```diff
@@ -4,29 +4,11 @@ require_relative '../deployment'
-SSHKit::Backend::Netssh.configure {|ssh| ssh.ssh_options = {paranoid: false}}
-
 unless CDO.daemon
   CDO.log.error 'This script must be run on daemon'
```
is this still true?

sureshc replied:
We were only using `sshkit` to connect to each web application server EC2 Instance and then invoke commands on the instance of Varnish installed on those servers, so we don't need to load and initialize it anymore.

cat5inthecradle reviewed (approved):
LGTM

What's the use case for `bin/flush_cache` after this?

There could definitely be unit tests on `cloudfront.invalidate_caches`. I don't have a great sense of how long that work would take though.

sureshc commented:
> LGTM
> 
> What's the use case for `bin/flush_cache` after this?
> 
> There could definitely be unit tests on `cloudfront.invalidate_caches`. I don't have a great sense of how long that work would take though.

We probably have documentation out there recommending running this script manually after making certain changes, that it's worth keeping this script around, but we should move towards having folks run the `ci:flush_cache` rake task from `production-daemon` in those cases.

sureshc commented:
Take another look?

I’ve renamed the task to clarify that it only flushes the CDN (HTTP) cache, deleted the `bin/flush_cache` script, made the `bin/flush_cloudfront_aws` script executable,  and I updated the Developer of the Day guide to reference `bin/flush_cloudfront_aws`.

Hamms reviewed (approved):
These changes look great to me! Agree that it would be nice to have unit tests in this space, but I'm comfortable releasing this change without them.
